pytorch_lightning/trainer/callback_hook.py

```diff
@@ -87,20 +89,38 @@ def on_validation_epoch_start(self):
         for callback in self.callbacks:
             callback.on_validation_epoch_start(self, self.lightning_module)
 
-    def on_validation_epoch_end(self):
+    def on_validation_epoch_end(self, outputs):
         """Called when the epoch ends."""
         for callback in self.callbacks:
-            callback.on_validation_epoch_end(self, self.lightning_module)
+            params = list(inspect.signature(callback.on_validation_epoch_end).parameters)
```

Comment:
Can you move this deprecation logic to a private function here and reuse for both methods?